lib/triangle.rb

```diff
@@ -1,3 +1,29 @@
 class Triangle
-  # write code here
+  def initialize(a, b, c)
+    if a == 0 && b == 0 && c ==0
+      raise TriangleError
+    elsif a < 0 || b < 0 || c < 0
+      raise TriangleError
+    elsif a + b <= c || c + b <= a || a + c <= b
+      raise TriangleError
+    else
+      @a = a
+      @b = b
+      @c = c
+    end
```

Comment:
I believe the intention is for this to happen in a method called `valid_triangle?`. This works here because you get an error, even if it's premature.